### ISSUE TITLE
New textbox(es) discussion and prototypes

### ISSUE THREAD

**Issue by enumag** [07th-mod/higurashi-patch-compiler#57]
We recently had a discussion on Discord about textboxes. While the current textbox is not bad, we want to try something even lighter and with less (if any) decorations.

@ItaloKnox Since you had some concrete ideas, can you post some first mockup of what you have in mind?

**Comment by ItaloKnox** [comment]
This is slightly different than what I promised, but I want to kickstart some discussion about changing (supposing it's actually possible) some of the elements of the textbox.

This textbox style favors less stuff in the screen and a more horizontal approach, it seems to be getting a little more popular lately in Japanese VNs.

First, some scenes directly from the game:

![NVIDIA_Share_2019-04-08_11-08-30](https://user-images.githubusercontent.com/4702556/55757634-cadb3280-5a2a-11e9-9660-f480a4009a3d.jpg)
![Photoshop_2019-04-08_18-11-54](https://user-images.githubusercontent.com/4702556/55757640-d0387d00-5a2a-11e9-8ed8-41b48eedf11e.jpg)

I tried to lower the text and move the name to the side, then added a very small dark gradient to make it pop a little:

![1](https://user-images.githubusercontent.com/4702556/55757708-0b3ab080-5a2b-11e9-976e-f2757f1e6321.jpg)
![3](https://user-images.githubusercontent.com/4702556/55757709-0bd34700-5a2b-11e9-8779-14ab23f77697.jpg)

This change gives the bottom some space to breathe and keeps a clean look, almost like subtitles. I also changed the font (and size - 22) to PT Sans since I believe MS Gothic is kinda... bad. Not sure if that would be possible in-game, though.

Here's the text *and* box in a white background:

![2](https://user-images.githubusercontent.com/4702556/55757860-8c924300-5a2b-11e9-8e44-77b82a52195e.jpg)

Obviously, the margins and everything else was measured by my eyes alone, so it's not a 1:1 thing. Supposing we decide to stick with something like this, I can take more than a few minutes to sketch something properly.

**Comment by ItaloKnox** [comment]
A small addendum: font rendering in-game and on Photoshop is slightly different, I don't intend to change the colors, outlines or anything like that. The only thing related to the fonts that I would *probably* change is the font size (slightly smaller) and the font itself. Everything else remains the same.

**Comment by enumag** [comment]
While your proposal does look nice, I don't think you considered everything:

- Sometimes there are 2-3 names on the screen. This would not fit.
- I'm uncertain if the engine even supports name positioning beside the text, we should be able to hack it though.
- There are cases with very long lines that are actually one voice and also one sentense making them impossible to split. We need to make sure these will fit. Even in current ADV mode I have to use lower font-size for some of these lines.
- Not sure if changing a font is even possible for us. I'll have to ask TellowKrinkle about this.

**Comment by tellowkrinkle** [comment]
We can change the font.  Do note that the current is *not* MS Gothic, but is Deja Vu Sans.

**Comment by ItaloKnox** [comment]
> 
> 
> While your proposal does look nice, I don't think you considered everything:
> 
>     * Sometimes there are 2-3 names on the screen. This would not fit.
> 
>     * I'm uncertain if the engine even supports name positioning beside the text. We should be able to hack it if needed though.
> 
>     * There are cases with very long lines that are actually one voice and also one sentense making them impossible to split. We need to make sure these will fit. Even in current ADV mode I have to use lower font-size for some of these lines.
> 
>     * Not sure if changing a font is even possible for us. I'll have to ask TellowKrinkle about this.

* I completely forgot about that, I guess that would make the design difficult.
* Yeah, I was aware that most likely wouldn't happen so easily.
* I believe this shouldn't really be an issue since the changes aren't that drastic. If it happened, I think that shifting the text some pixels above might make enough room for a new line.
* Well, looks like we can but it's not obligatory. I forgot we used Deja Vu Sans as @tellowkrinkle said above, so it's partially a misunderstanding. It should work well enough if trying a new font would be a bother.

**Comment by enumag** [comment]
@tellowkrinkle @idealpersona would it be possible to change the black outline of the font if needed? (To make the letters more readable when a really light textbox is used.)

**Comment by tellowkrinkle** [comment]
Not sure, I can look into it later.  There's a pretty good chance that it will be changeable per text box (and not for subsections) if it is easily changeable though.
Also isn't black the outline color you would want for a light text box?

**Comment by ItaloKnox** [comment]
[Here is the upload](https://lithi.io/file/en4i.png) of the minimal textbox I made.

I also made a new one using the same base textbox plus some texturing with brushes. [Here it is](https://lithi.io/file/I6Ad.png).

That's how it should look "in-game":
![rena](https://user-images.githubusercontent.com/4702556/56181258-0f466f80-5fe3-11e9-9ea1-cb9ac8af89e4.jpg)

Those were designed to be used with the slider set to max just like we recommend doing with our current textbox (that is much darker). Or at least I believe so.

@enumag 
The details you requested:
Font size: 22
Font used in the mockup: PT Sans

Just as I said before, those are definitely *not* requirements. It might look totally different in-game due to the font rendering. We can keep the current font and do a few changes to the size until we settle with something we're all okay with, changing the font is very minor.
